internal/schemautil/service.go

```diff
@@ -864,3 +904,17 @@ func DatasourceServiceRead(ctx context.Context, d *schema.ResourceData, m interf
 
 	return diag.Errorf("common %s/%s not found", projectName, serviceName)
 }
+
+func getContactEmailListForAPI(d *schema.ResourceData, field string) *[]aiven.ContactEmail {
+	var results []aiven.ContactEmail
+	valuesInterface, ok := d.GetOk(field)
+	if ok && valuesInterface != nil {
+		for _, emailInterface := range valuesInterface.(*schema.Set).List() {
+			results = append(results, aiven.ContactEmail{Email: emailInterface.(map[string]interface{})["email"].(string)})
+		}
+	}
+	if results == nil {
+		return &[]aiven.ContactEmail{}
+	}
+	return &results
```

Comment:
You can safely remove 916-918. `results` is the same as that part.

Reply:
Ah, you see, `*[]` doesn't make sense. Cause you always return a pointer to a slice. Hence it sends `null`, because it has a type, but not a length. https://go.dev/play/p/FJLtc1sE-xf
You need to return `nil` to make your `if` check make sense.
I believe that the API just ignores `null` and that's why it works.

So you can do this:
```go
results := make([]aiven.ContactEmail, 0)
valuesInterface, ok := d.GetOk(field)
if ok {
    ...
}
return &results
```

It will always send an array. Works exactly like if you replace `*[]email` with `[]email` in the client.